src/boot/store.js

```diff
@@ -118,24 +133,104 @@ const migrations: { [string]: (GlobalState) => GlobalState } = {
 };
 
+/**
+ * Return a list of Redux middleware objects to use in our Redux store.
+ *
+ * See Redux docs on its "middleware API":
+ *   https://redux.js.org/api/applymiddleware/
+ */
+function listMiddleware() {
+  const result = [
+    // Allow us to cause navigation by dispatching Redux actions.
+    // See docs: https://github.com/react-navigation/redux-helpers
+    createReactNavigationReduxMiddleware('root', getNav),
+
+    // Delay ("buffer") actions until a REHYDRATE action comes through.
+    // After dispatching the latter, this will go back and dispatch
+    // all the buffered actions.  See docs:
+    //   https://github.com/rt2zz/redux-action-buffer
+    createActionBuffer(REHYDRATE),
+
+    // Handle the fancy "thunk" actions we often use, i.e. async
+    // functions of `dispatch` and `state`.  See docs:
+    //   https://github.com/reduxjs/redux-thunk
+    thunkMiddleware,
+  ];
+
+  if (config.enableReduxLogging) {
+    result.push(
+      // Log each action to the console -- often handy in development.
+      // See upstream docs:
+      //   https://github.com/LogRocket/redux-logger
+      // and ours:
+      //   https://github.com/zulip/zulip-mobile/blob/master/docs/howto/debugging.md#redux-logger
+      createLogger({
+        duration: true,
+        // Example options to add for more focused information, depending on
+        // what you're investigating; see docs/howto/debugging.md (link above).
+        //   diff: true,
+        //   predicate: (getState, action) => action.type === 'MESSAGE_FETCH_COMPLETE',
+      }),
+    );
+  }
+
+  return result;
+}
+
+/**
+ * The Redux store.  We store nearly all application data here.
+ *
+ * For discussion, see:
+ *  * docs/architecture.md
+ *  * docs/architecture/realtime.md
+ *  * docs/background/recommended-reading.md
+ */
 const store: Store<GlobalState, Action> = createStore(
   rootReducer,
   undefined,
   compose(
+    // Invoke redux-persist-migrate with our migrations.
     createMigration(migrations, 'migrations'),
-    applyMiddleware(...middleware),
+
+    // Various middleware; see `listMiddleware`.
+    applyMiddleware(...listMiddleware()),
+
+    // Handle all the boring parts of a REHYDRATE action from redux-persist,
+    // where the live state just gets filled in with the corresponding parts
+    // of the just-loaded state from disk.  See upstream docs:
+    //   https://github.com/rt2zz/redux-persist/tree/v4.10.2#autorehydrateconfig
     autoRehydrate(),
   ),
 );
 
+/**
+ * The config options to pass to redux-persist.
+ *
+ * See upstream docs:
+ *   https://github.com/rt2zz/redux-persist/tree/v4.10.2
+ *
+ * (Why v4?  We studied v5 and determined it doesn't make sense for us to
+ * use because its design choices on migrations don't support some important
+ * kinds of migrations.  See commit 081a09806, PR #2762, and this discussion:
+ *   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/redux-persist.20v5/near/624465
+ * all from 2018-08.)
```

Comment:
Also e5409c578.

Reply:
Ah, looks like https://github.com/zulip/zulip-mobile/commit/e5409c57833c9239b98482f4a8ff4a56c730f85d has all three of these references, so I bet we can just refer to https://github.com/zulip/zulip-mobile/commit/e5409c57833c9239b98482f4a8ff4a56c730f85d by itself.